cpu/atmega_common/periph/timer.c

```diff
@@ -136,9 +137,34 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
         return -1;
     }
 
+    unsigned state = irq_disable();
     ctx[tim].dev->OCR[channel] = (uint16_t)value;
     *ctx[tim].flag &= ~(1 << (channel + OCF1A));
     *ctx[tim].mask |= (1 << (channel + OCIE1A));
+    irq_restore(state);
+
+    return 0;
+}
+
+int timer_set(tim_t tim, int channel, unsigned int timeout)
+{
+    if (channel >= TIMER_CHANNELS) {
+        return -1;
+    }
+
+    unsigned state = irq_disable();
+    unsigned absolute = ctx[tim].dev->CNT + timeout;
+    ctx[tim].dev->OCR[channel] = absolute;
+    *ctx[tim].mask |= (1 << (channel + OCIE1A));
+    if (absolute <= ctx[tim].dev->CNT) {
```

Comment:
Isn't this always true if `CNT+timeout` overflow 16bit? Eg, on `CNT==0xfff0`, `timeout=100`, then `absolute` would be 84, `absolute <= ctx[tim].dev->CNT` would be true but there's still time left?